spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/AbstractSpannerIntegrationTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseId;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerPersistentEntity;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerPersistentEntityImpl;
+import org.springframework.cloud.gcp.data.spanner.test.domain.Trade;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.data.util.TypeInformation;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assume.assumeThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class provides the foundation for the integration test framework for Spanner.
+ * Its responsibilities:
+ * <ul>
+ *   <li>initializes the Spring application context</li>
+ *   <li>sets up the database schema</li>
+ *   <li>manages table suffix generation so that parallel test cases don't collide</li>
+ * </ul>
+ * <p>
+ * Prerequisites for running integration tests:
+ *
+ * For Spanner integration tests, you will need to have an instance predefined, everything
+ * else is generated. The instance by default is "integration-instance", which can be
+ * overriden in <code>src/test/resources/application-test.properties</code>, the property:
+ * <code>test.integration.spanner.instance</code>
+ * </p>
+ * <p>
+ * Within the <code>integration-instance</code> instance, the tests rely on a single database for
+ * tests, <code>integration-db</code>. This is automatically created, if doesn't exist.
+ * The tables are generated to have a unique postfix, which is updated on the entity
```

Comment:
Shouldn't we change this `postfix` to a `suffix` to avoid some confusions? :wink:

Reply:
:+1: hehe